### ISSUE TITLE
Syntax error in mysql_user under Python 2.4

### ISSUE THREAD

**Issue by vanne** [ansible/ansible#1577]
I'm getting

```
fatal: [1.2.3.4] => failed to parse:   File "/var/tmp/ansible-1352467343.1-175910241019560/mysql_user", line 177
    privileges = ['ALL' if x=='ALL PRIVILEGES' else x for x in privileges]
                         ^
SyntaxError: invalid syntax
```

Works fine on a managed node with 2.6, not so when it has 2.4.
The line is [this one](https://github.com/ansible/ansible/blob/devel/library/mysql_user#L177)

**Comment by mpdehaan** [comment]
Yeah, this code is heinous ugly.

The list comprehension should be written such that the "if/else" happens after the for at least, though I need to check if 2.4 supports the else

There is nothing wrong with this NOT being a list comprehension also.

**Comment by mpdehaan** [comment]
fix pushed in  8647aec ... I don't use MySQL so this is VERY lightly tested, let me know if this works for you

**Comment by vanne** [comment]
Sorry for the late reply.
Works for me, yes. Thank you.
